phases/00-setup-and-tooling/12-debugging-and-profiling/code/debug_tools.py

```
@@ -224,26 +229,40 @@ def demo_gradient_health():
-        print("  No GPU available. Skipping GPU memory demo.")
-        print("  On a GPU machine, torch.cuda.memory_summary() shows:")
-        print("    - Allocated memory per block size")
-        print("    - Cached (reserved) memory")
-        print("    - Peak memory usage")
+    if torch.cuda.is_available():
+        print(f"  GPU: {torch.cuda.get_device_name(0)}")
+        print(f"  Allocated: {torch.cuda.memory_allocated() / 1e6:.1f} MB")
+        print(f"  Cached: {torch.cuda.memory_reserved() / 1e6:.1f} MB")
+
+        large_tensor = torch.randn(10000, 10000, device="cuda")
+        print(f"  After 10k x 10k tensor:")
```

Ruff (0.15.15)

[error] f-string without any placeholders

Remove extraneous `f` prefix

(F541)